src/common/components/clusterConfiguration/utils.ts

```diff
@@ -105,6 +106,44 @@ export const isAdvNetworkConf = (
   selectServiceNetworkCIDR(cluster) !== defaultNetworkSettings.serviceNetworkCidr ||
   (Boolean(cluster.networkType) && cluster.networkType !== defaultNetworkType);
 
+export const canSelectNetworkTypeSDN = (isSNO: boolean, isIPv6 = false) => {
+  return !(isSNO || isIPv6);
+};
+
+export const getDefaultNetworkType = (isSNO: boolean, isIPv6 = false) => {
+  return isSNO || isIPv6 ? NETWORK_TYPE_OVN : NETWORK_TYPE_SDN;
+};
+
+export const canBeDualStack = (subnets: HostSubnets) =>
+  subnets.some((subnet) => Address4.isValid(subnet.subnet)) &&
+  subnets.some((subnet) => Address6.isValid(subnet.subnet));
+
+const areNetworksDualStack = (
+  networks: (MachineNetwork | ClusterNetwork | ServiceNetwork)[] | undefined,
+) =>
+  networks &&
+  networks.length > 1 &&
+  Address4.isValid(networks[0].cidr || '') &&
+  Address6.isValid(networks[1].cidr || '');
+
+export const isDualStack = ({
+  machineNetworks,
+  clusterNetworks,
+  serviceNetworks,
+}: Pick<Cluster, 'machineNetworks' | 'clusterNetworks' | 'serviceNetworks'>) =>
+  areNetworksDualStack(machineNetworks) &&
+  areNetworksDualStack(clusterNetworks) &&
+  areNetworksDualStack(serviceNetworks);
+
+export const isSubnetInIPv6 = ({
```

Comment:
```suggestion
export const isSubnetInIPv6 = ({
  clusterNetworkCidr,
  machineNetworkCidr,
  serviceNetworkCidr,
}: Pick<Cluster, 'clusterNetworkCidr' | 'machineNetworkCidr' | 'serviceNetworkCidr'>) =>

I just noticed this TS error. IMO we should make sure that Eslint catches these errors as our code will be more robust and we'll catch errors automatically. @jkilzi I'll try to fix this so tht linting fails in these cases.
```

Reply:
The issue is one that we have excluded, `@typescript-eslint/no-unsafe-call` and to fix the errors won't be very quick, so for now we won't have them fixed.